kubejs/server_scripts/enigmatica/kubejs/functions.js

```diff
@@ -46,6 +46,16 @@ function wrapArray(array) {
     return utils.listOf(array).toArray();
 }
 
+function getStrippedLog(value) {
+    let strippedLog = air;
+    for (var i = 0; i < buildWoodVariants.length; i++) {
+        if (buildWoodVariants[i].logBlock === value) {
+            strippedLog = buildWoodVariants[i].logBlockStripped;
+        }
+    }
+    return strippedLog;
+}
```

Comment:
May I suggest a different approach?

```js
function getStrippedLogFrom(logBlock) {
    buildWoodVariants.find((wood) => {
        if (wood.logBlock == logBlock) {
            return wood.logBlockStripped;
        }
    });
}

Reply:
I think there is a return missing in `getStrippedLogFrom` with your method @NillerMedDild

Reply:
ah true, the air part is needed

Reply:
yeah some trees don't have a stripped block, like fruit trees. so I need air as a default. It's more the actual search that I don't like. Seems like this being an array of objects I shouldn't have to iterate on it...

Reply:
which, of course, you're showing
lol

Reply:
Find still iterates it I'm afraid :/

Reply:
Yeah, not getting any matches with this, unfortunately. I'm ending up with Air as a result for every request.

```js
function getStrippedLogFrom(logBlock) {
    buildWoodVariants.find((wood) => {
        console.log(`Looking for: ${logBlock}`);
        if (wood.logBlock == logBlock) {
            console.log(`Found: ${wood.logBlockStripped}`);
            return wood.logBlockStripped;
        } else {
            console.log(`Not Found: ${logBlock}`);
            return air;
        }
    });
}
```
snippet of the log:
```
[15:49:31] [Server thread/INFO]: server_scripts:enigmatica/kubejs/functions.js:61: Looking for: byg:cika_log
[15:49:31] [Server thread/INFO]: server_scripts:enigmatica/kubejs/functions.js:66: Not Found: byg:cika_log
```

Reply:
Oh alright, sorry about that 😅

Reply:
Actually.. You're only checking the first variant.

Reply:
```js
function getStrippedLogFrom(logBlock) {
    buildWoodVariants.find((wood) => {
        console.log(`Looking for: ${logBlock}`);
        if (wood.logBlock == logBlock) {
            console.log(`Found: ${wood.logBlockStripped}`);
            return wood.logBlockStripped;
        }
    });
    console.log(`Not Found: ${logBlock}`);
    return air;
}

Reply:
Oooooh.